### DISCUSSION TITLE
how can I send out request with special characters without urlencode?

### DISCUSSION THREAD

**Discussion by DorisQQ2019** [Kong/unirest-java#432] [General]
Hello, this time I get "Exception in thread "main" kong.unirest.UnirestException: java.lang.IllegalArgumentException: Illegal character in query at index 33." when I send out request for target url like "http://target/api?keyword=name:A`B`C`D
". I guess this is because the special character "`". However when I urlencode it before sending out, I cannot get the right response. I saw urllib2.urlopen(request) in python works exactly what I want. Is here any approach I can send out the request successfully and get the expected response?

**Comment by ryber** [comment]
` isn't a valid URL character, this isn't even a error from Unirest, it's from Java, the only way around it is to encode the character

**Reply by DorisQQ2019** [reply]
Search in website, got following instructions: https://en.wikipedia.org/wiki/Percent-encoding#Character_data
https://docs.oracle.com/javase/7/docs/api/java/net/URI.html
it seems the problem comes from the different URI encoding approach between RFC 2396 and current standard RFC 3987.
